src/otx/engine/engine.py

```diff
@@ -743,10 +743,14 @@ def _build_lightning_module(
         class_module, class_name = LITMODULE_PER_TASK[self.task].rsplit(".", 1)
         module = __import__(class_module, fromlist=[class_name])
         lightning_module = getattr(module, class_name)
-        return lightning_module(
-            otx_model=model,
-            optimizer=optimizer,
-            scheduler=scheduler,
-            torch_compile=False,
-            metric=metric,
-        )
+
+        lightning_kwargs = {
+            "otx_model": model,
+            "optimizer": optimizer,
+            "scheduler": scheduler,
+            "torch_compile": False,
+        }
+        if metric:
+            lightning_kwargs["metric"] = metric
```

Comment:
You should change typing at L730 in this file to MetricCallable

Reply:
Done, https://github.com/openvinotoolkit/training_extensions/pull/2964/commits/3331c3032f6daad27eb5088f951abb1133a34d8b